pkg/apis/core/validation/validation.go

```diff
@@ -5332,3 +5333,32 @@ func ValidateProcMountType(fldPath *field.Path, procMountType core.ProcMountType
 		return field.NotSupported(fldPath, procMountType, []string{string(core.DefaultProcMount), string(core.UnmaskedProcMount)})
 	}
 }
+
+// validateTopologySpreadConstraints validates given TopologySpreadConstraints.
+func validateTopologySpreadConstraints(constraints []core.TopologySpreadConstraint, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	supportedRespModes := sets.NewString(string(core.DoNotSchedule), string(core.ScheduleAnyway))
+	// tuple {topologyKey, whenUnsatisfiable} denotes one kind of constraint
+	allConstraintKinds := sets.String{}
+	for i, constraint := range constraints {
+		subFldPath := fldPath.Index(i)
+		if val := constraint.MaxSkew; val <= 0 {
```

Comment:
I think generally folks make separate functions rather than validate a bunch of things inside a loop like this.

Reply:
done